pkg/controlplane/controller.go

```diff
@@ -101,6 +101,10 @@ func (c *completedConfig) NewBootstrapController(legacyRESTStorage corerest.Lega
 		}
 	}
 
+	if c.GenericConfig.PublicPort > 0 {
```

Comment:
nit, but this override may be better close to Line88 , that is where we get the `publicServicePort`var